custom_components/acwd/helpers.py

```
@@ -1,11 +1,43 @@
+    """
+    if "d" not in result:
+        raise ValueError(
+            f"Unexpected API response from {endpoint}: missing 'd' property"
+        )
+    try:
+        return json.loads(result["d"])
+    except json.JSONDecodeError as e:
+        snippet = result["d"][:200]
+        raise ValueError(
+            f"Failed to parse API response from {endpoint}: {e} (got: {snippet!r})"
+        ) from e
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)